File: phSolver/common/gtnods.f
      subroutine gtnods

      include "common.h"
      include "mpif.h"
      include "auxmpi.h"

!SCATTER      dimension irecvcount(numpe), numvec(numpe)

      if(numpe > 1) then
         irecvcount = 1
         numvec = nshg0
         call MPI_ALLREDUCE (numvec, nshgt, irecvcount,
     &                 MPI_INTEGER, MPI_SUM, MPI_COMM_WORLD, ierr)
c         call MPI_REDUCE_SCATTER (numvec, nshgt, irecvcount,
c     &                 MPI_INTEGER, MPI_SUM, MPI_COMM_WORLD, ierr)
      else
         nshgt = nshg0
      endif
     
      if (myrank .eq. master) then
         write(6,*) 'Total number of modes = ',nshgt
      endif
      
      return
      end

File: phSolver/common/gtnods.c
#include <stdio.h>
#include <mpi.h>

#include "common_c.h"

#include <FCMangle.h>  
#define gtnods FortranCInterface_GLOBAL_(gtnods, GTNODS) 

void gtnods()
{

	if(workfc.numpe > 1) {

		int irecvcount, ierr;
		long long numvec, loc_nshgt;

                irecvcount = 1;
		numvec = (long long) newdim.nshg0;

//		printf("Local number of modes = %ld %d %d\n",numvec,newdim.nshg0, sizeof(newdim.nshg0));

		ierr = MPI_Allreduce(&numvec, &loc_nshgt, irecvcount,
                                MPI_LONG_LONG, MPI_SUM, MPI_COMM_WORLD);

		newdim.nshgt = loc_nshgt;	
	}
	else {
		newdim.nshgt = (long long) conpar.nshg;
	}

	if (workfc.myrank == workfc.master) {
		printf("Total number of modes = %ld\n",newdim.nshgt);
 	}

}
